rebasehelper/plugins/spec_hooks/replace_old_version.py

```diff
@@ -73,7 +74,9 @@ def _replace(cls, line, old, new, replace_with_macro=False):
         """
         if cls._is_local_source(line):
             return line
-        return line.replace(old, '%{version}' if replace_with_macro else new)
+        pattern = re.compile(r'([/\-\s]){}([/.\-\s])'.format(re.escape(old)))
```

Comment:
This could be compiled outside the loop (old version is already known) to improve performance.